homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -1 +1,81 @@
 """Support for Xiaomi Miio."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_TOKEN,
+)
+
+from miio import DeviceException, gateway
+
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'xiaomi_miio'
+
+CONF_GATEWAYS = "gateways"
+KEY_GATEWAY_DEVICE = "device"
+KEY_GATEWAY_INFO = "info"
+
+DEFAULT_ALARM_NAME = "Xiaomi Gateway Alarm"
+
+GATEWAY_CONFIG = vol.Schema(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_TOKEN): vol.All(cv.string, vol.Length(min=32, max=32)),
+        vol.Optional(CONF_NAME, default=DEFAULT_ALARM_NAME): cv.string,
+    }
+)
+
+CONFIG_SCHEMA = vol.Schema(
```

Comment:
We recently accepted ADR10 which makes this legacy already.

See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

Please refactor to use a config flow and config entries instead.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
well at least I made a start to centrilize the configuration, I will continue tomorrow and try to update it to the config flow.

The Xiaomi_aqara platform uses this same way of setting up the platform so that integration will probably also need refactoring.

@MartinHjelmare Could the HomeAssistant Dev Docs be updated to reflect these decisions, I got my info from:
https://developers.home-assistant.io/docs/creating_component_generic_discovery/
Under "Loading platforms when configured via configuration.yaml"
The example mentioned there is also depricated in that case:
https://github.com/home-assistant/example-custom-config/tree/master/custom_components/example_load_platform/

Reply:
Some integrations are still allowed to use config yaml, so we still need that part of the docs. We should probably make it clear when that is an option, though.